Source/JavaScriptCore/runtime/ClonedArguments.cpp

```diff
@@ -49,7 +49,11 @@ ClonedArguments* ClonedArguments::createEmpty(VM& vm, Structure* structure, JSFu
 
     if (UNLIKELY(structure->mayInterceptIndexedAccesses() || structure->storedPrototypeObject()->needsSlowPutIndexing())) {
         if (!butterfly) {
-            butterfly = createArrayStorageButterfly(vm, nullptr, structure, length, vectorLength);
+            butterfly = tryCreateArrayStorageButterfly(vm, nullptr, structure, length, vectorLength);
+            if (UNLIKELY(!butterfly)) {
+                throwOutOfMemoryError(structure->globalObject(), scope);
```

Comment:
What happen when it is throwing an error from `operationCreateClonedArgumentsDuringExit` in OSR exit?

Reply:
Hmmm, that is an interesting question.  However, it is a pre-existing condition as `ClonedArguments::createEmpty` was already able to throw an OOME before this.  See the else case.

I think an OOME during OSR exit is catastrophic, and we currently already crash when we try to put arguments into the ClonedArguments.  I'll add a RELEASE_ASSERT in `operationCreateClonedArgumentsDuringExit` to make it explicit that `ClonedArguments::createEmpty` must not return nullptr here.